rust/db_handling/src/identities.rs

```diff
@@ -1104,3 +1210,200 @@ pub fn prepare_derivations_import(
         &derivations,
     ))
 }
+
+#[cfg(feature = "signer")]
+pub(crate) fn prepare_secret_key_for_export(
+    multisigner: &MultiSigner,
+    full_address: &str,
+    pwd: Option<&str>,
+) -> Result<[u8; 32]> {
+    match multisigner {
+        MultiSigner::Ed25519(public) => {
+            let ed25519_pair =
+                ed25519::Pair::from_string(full_address, pwd).map_err(Error::SecretStringError)?;
+            if public != &ed25519_pair.public() {
+                return Err(Error::WrongPassword);
+            }
+            Ok(*ed25519_pair.seed())
+        }
+        MultiSigner::Sr25519(public) => {
+            let (sr25519_pair, seed) = sr25519::Pair::from_string_with_seed(full_address, pwd)
+                .map_err(Error::SecretStringError)?;
+            if public != &sr25519_pair.public() {
+                return Err(Error::WrongPassword);
+            }
+            Ok(seed.ok_or_else(|| Error::NoSeedForKeyPair {
+                multisigner: multisigner.clone(),
+            })?)
+        }
+        MultiSigner::Ecdsa(public) => {
+            let ecdsa_pair =
+                ecdsa::Pair::from_string(full_address, pwd).map_err(Error::SecretStringError)?;
+            if public != &ecdsa_pair.public() {
+                return Err(Error::WrongPassword);
+            }
+            Ok(ecdsa_pair.seed())
+        }
+    }
+}
+
+/// Prepare **secret** key export screen struct [`MKeyDetails`].
+///
+/// For QR code the secret key information is put in format
+/// `secret:0x{}:{network genesis hash}` transformed into bytes, to be
+/// compatible with `polkadot-js` interface.
+///
+/// There is no direct mentioning of [`Encryption`] in the QR code. It is
+/// that user knows the key in which network and with what encryption they are
+/// exporting.
+///
+/// The QR code here contains sensitive information, and is made with special
+/// coloration, so that the difference with safe QR codes is immediately visible
+/// on screen.
+#[cfg(feature = "signer")]
+pub fn export_secret_key(
```

Comment:
This function seems quite massive, do we have any linting rules on maximum `fn` size? 
Should we extract some of the functionality, like `let mut qr = {` into private function?

Reply:
Agree, in fact I thought that `clippy` should have caught it

Reply:
Fair enough